### PR TITLE
Doc improvement for comparison

### PR DESCRIPTION
**Pull request by samuelgarcia**
* more plot in gallery examples.

### CODE REVIEW

File: examples/modules/comparison/plot_2_compare_multiple_sorters.py [hunk 1]
```diff
@@ -49,6 +50,12 @@
 print(mcmp.comparisons[1].sorting1, mcmp.comparisons[1].sorting2)
 mcmp.comparisons[0].get_mapped_sorting1().get_mapped_unit_ids()
 
+
+#############################################################################
+# The global multi caomparison can be visualize with this graph
```
Comment (alejoe91) [inline examples/modules/comparison/plot_2_compare_multiple_sorters.py]: visualized

File: examples/modules/comparison/plot_1_compare_two_sorters.py [hunk 2]
```diff
@@ -40,6 +42,18 @@
 cmp_KL_MS4 = sc.compare_two_sorters(sorting1=sorting_KL, sorting2=sorting_MS4, 
                                                sorting1_name='klusta', sorting2_name='ms4')
 
+#############################################################################
+# We can check the agreement matrix to check the matching.
```
Comment (alejoe91) [inline examples/modules/comparison/plot_1_compare_two_sorters.py]: to check --> to inspect

File: examples/modules/comparison/plot_1_compare_two_sorters.py [hunk 3]
```diff
@@ -40,6 +42,18 @@
...
+
+sw.plot_agreement_matrix(cmp_KL_MS4)
+
+#############################################################################
+# Some usefull internal dataframes help to check the match and count like **match_event_count**
```
Comment (alejoe91) [inline examples/modules/comparison/plot_1_compare_two_sorters.py]: usefull -> useful
and counts. For ecample **match_ ...

File: examples/modules/comparison/plot_3_compare_sorter_with_ground_truth.py [hunk 4]
```diff
@@ -35,6 +36,17 @@
 
 cmp_gt_MS4 = sc.compare_sorter_to_ground_truth(sorting_true, sorting_MS4, exhaustive_gt=True)
 
+
+##############################################################################
+# To have an overview of the match we can use the matrix agreement unordered
```
Comment (alejoe91) [inline examples/modules/comparison/plot_3_compare_sorter_with_ground_truth.py]: unordered agreement matrix

File: examples/modules/comparison/plot_3_compare_sorter_with_ground_truth.py [hunk 5]
```diff
@@ -73,6 +85,12 @@
 ax1 = sns.swarmplot(data=perf2, x='measurement', y='value', ax=ax1)
 ax1.set_xticklabels(labels=ax1.get_xticklabels(), rotation=45)
 
+##############################################################################
+# The confusion matrix is also a good summary of the score
+# it have the same shape as agreement matrix but have an extra columns for FN
+# and an extra row for FP
+
```
Comment (alejoe91) [inline examples/modules/comparison/plot_3_compare_sorter_with_ground_truth.py]: # The confusion matrix is also a good summary of the score as it has
# the same shape as agreement matrix, but it contains an extra column for FN
# and an extra row for FP

**Comment by samuelgarcia** [comment]
@alejoe91 @colehurwitz @magland @mhhennig 
Hi all,
I have added a new example "plot_5_comparison_sorter_weaknesses.py" in the gallery.

It show that redundant/bad/well detected are concept not so clear!!!!!

Alessio have started a PR here https://github.com/SpikeInterface/spikecomparison/pull/19

But this is just a begenning of work.

Could you have a look ?


Sam

**Comment by samuelgarcia** [comment]
This should pass test if this is merge:
https://github.com/SpikeInterface/spikesorters/pull/66
but MS4 hangs.

**Comment by alejoe91** [comment]
Merge?

**Comment by samuelgarcia** [comment]
yesssssssssssssssssssssssssssssssssssssss

**Pull request merged**